src/PagesProgressBar.tsx

```diff
@@ -52,7 +52,7 @@ export const PagesProgressBar = React.memo(
 
       const handleRouteStart = (url: string) => {
         const targetUrl = new URL(url, location.href);
-        const currentUrl = new URL(Router.route, location.href);
+        const currentUrl = new URL(Router.asPath, location.href);
```

Comment:
i've observed that this still doesn't work when the application is configured to have a `basePath`, and i suspect it may not work if the app has a locale in the url as well

this is because the `targetUrl` will have the configured `basePath` and current locale (if non-default) encoded into the pathname of its url, but the `currentUrl` will not because `Router.asPath` omits both `basePath` and locale (see docs: https://nextjs.org/docs/pages/api-reference/functions/use-router#router-object), meaning the equality check will not match

ideally this method could be updated to handle the `basePath` configuration, as well as the existence (or lack thereof) of a locale in the pathname - all of this information should be readable from the `Router` object to my knowledge, so hopefully not too complex

handling `basePath` may be as simple as just prefixing it to `Router.asPath` - eg:
```suggestion
        const currentUrl = new URL(`${Router.basePath}${Router.asPath}`, location.href);
```

but handling non-default locales may be more complex 🤔

Reply:
in fact, is it insufficient to just use `location.href` directly for the current URL? do we need to read anything from the router config? `location.href` should have everything we need already encoded (e.g. `basePath`, locale, etc.) - is there a particular reason for interpreting the URL from the router instead?
```suggestion
        const currentUrl = new URL(location.href);
```

Reply:
Those are great observations. I actually just noticed that `location.href` is used in the `AppProgressBar` for this comparison:

https://github.com/Skyleen77/next-nprogress-bar/blob/50aae80d4207addea77e9bb082cdee49b44ab283/src/AppProgressBar.tsx#L132

I've applied your suggestion!